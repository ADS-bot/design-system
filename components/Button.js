import PropTypes from 'prop-types'
import React from 'react'
import { Button as RButton } from 'rebass/styled-components'

export const Button = props => {
  let variant = 'primary'
  if (props.hollow == true) {
    variant = 'hollow'
  }
  if (props.inverted == true) {
    variant = 'inverted'
  }
  if (props.variant !== undefined) {
    variant = props.variant
  }
  return <RButton
    {...props}
    variant={variant}
    sx={{
      borderWidth: props.border
    }}
  />
}

Button.propTypes = {
  fontSize: PropTypes.number,
  inverted: PropTypes.bool,
  hollow: PropTypes.bool,
  border: PropTypes.number,
  variant: PropTypes.string
}

Button.defaultProps = {
  fontSize: 1,
  border: 2
}

export default Button
