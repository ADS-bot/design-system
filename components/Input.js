import React from 'react'

import { space, width, fontSize, color } from 'styled-system'
import styled from 'styled-components'

export const Input = styled.input`
  $space;
  $width;
  $fontSize;
  $color;

  font-family: inherit;
  font-size: 1rem;
  display: 'block';
  vertical-align: 'middle';

  background-color: transparent;

  border: none;
  border-bottom: 1px solid ${props => props.theme.colors.gray2};
  border-radius: 0;
  margin: 0;
  margin-bottom: 15px;
  padding: 0;

  outline: none;
  width: 100%;
  height: 2rem;

  box-shadow: none;
  box-sizing: content-box;
  transition: all .3s;

  &:focus:not([readonly]) {
    border-bottom: 1px solid ${props => props.theme.colors.base};
    box-shadow: 0 1px 0 0 ${props => props.theme.colors.gray1};
  }
`
export default Input
