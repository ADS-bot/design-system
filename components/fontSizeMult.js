/* global require */
import {
  system
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return system({
    'fontSize': {
      property: propName,
      scale: 'fontSizes',
      transform: n => n * k,
      defaultScale: defaultFontSizes
    }
  })
}

export default fontSizeMult
