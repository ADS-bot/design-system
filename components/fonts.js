import { injectGlobal } from 'styled-components'

import CharterRegular from '../fonts/Charter-Regular.woff'
import CharterBold from '../fonts/Charter-Bold.woff'

import FiraSansLight from '../fonts/FiraSans-Light.woff'
import FiraSansRegular from '../fonts/FiraSans-Regular.woff
import FiraSansBold from '../fonts/FiraSans-Bold.woff'

import SourceCodeProBold from '../fonts/SourceCodePro-Bold.woff'
import SourceCodeProRegular from '../fonts/SourceCodePro-Regular.woff'


injectGlobal`
  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansLight}') format('woff');
    font-style: normal;
    font-weight: 300;
  }

  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansRegular}') format('woff');
    font-style: normal;
    font-weight: 400;
  }

  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansSemiBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }

  @font-face {
    font-family: "Source Code Pro";
    src: url('${SourceCodeProRegular}') format('woff');
    font-style: normal;
    font-weight: 400;
  }

  @font-face {
    font-family: "Source Code Pro";
    src: url('${SourceCodeProBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }

  @font-face {
    font-family: "Charter";
    src: url('${CharterRegular}') format('woff');
    font-style: normal;
    font-weight: 300;
  }

  @font-face {
    font-family: "Charter";
    src: url('${CharterBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }
`
