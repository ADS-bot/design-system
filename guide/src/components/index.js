import  ColorPalette from './ColorPalette'

const GuideComponents = Object.assign({}, { ColorPalette })
export default GuideComponents
