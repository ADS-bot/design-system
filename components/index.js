import {
  Flex,
  Box,
  Link,
} from 'rebass/styled-components'

import { Select, Label } from '@rebass/forms/styled-components'

import { ThemeProvider as Provider } from 'styled-components'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Flex,
  Box,
  Link,
} from 'rebass/styled-components'

export { Select, Label } from '@rebass/forms/styled-components'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Input,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, {
  Provider,
  Box,
  Flex,
  Link,
  Select,
  Label,
  colors,
  theme
})

export default OONIComponents
