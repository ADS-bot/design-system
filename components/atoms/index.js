import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import Container from './Container'
import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  TwitterShareButton,
  FacebookShareButton,
  Container
}

export default atoms
