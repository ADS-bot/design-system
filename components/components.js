import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */
import Button from './Button'
export { default as Button } from './Button'
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Container = atoms.Container

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Circle,
  Close,
  Code,
  Column,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Lead,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass/styled-components'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Lead,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Small,
  Star,
  Sticky,
  Subhead,
  TabItem,
  Tabs,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
