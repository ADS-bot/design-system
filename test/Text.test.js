/* global describe, test, expect */
import React from 'react'
import { renderJSON, renderShallowJSON } from './index'
import { Text } from '../components'

describe('Text', () => {
  test('renders', () => {
    const json = renderJSON(
      <Text />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with prop fontWeight', () => {
    const json = renderShallowJSON(
      <Text fontWeight={500} />
    )

    expect(json).toMatchSnapshot()
  })

  test('renders with prop fontSize', () => {
    const json = renderShallowJSON(
      <Text fontSize={18} />
    )

    expect(json).toMatchSnapshot()
  })

  test('renders with prop textAlign', () => {
    const json = renderShallowJSON(
      <Text textAlign={'center'} />
    )

    expect(json).toMatchSnapshot()
  })

  test('renders with prop width', () => {
    const json = renderShallowJSON(
      <Text width={1} />
    )

    expect(json).toMatchSnapshot()
  })
})
