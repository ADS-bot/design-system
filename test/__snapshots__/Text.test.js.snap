// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  class="c0"
/>
`;

exports[`Text renders with prop fontSize 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 18px;
}

<div
  class="c0"
  font-size="18"
/>
`;

exports[`Text renders with prop fontWeight 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-weight: 500;
}

<div
  class="c0"
  font-weight="500"
/>
`;

exports[`Text renders with prop textAlign 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  text-align: center;
}

<div
  class="c0"
/>
`;

exports[`Text renders with prop width 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
}

<div
  class="c0"
  width="1"
/>
`;
